python/cudf/cudf/core/column/lists.py

```diff
@@ -453,10 +454,59 @@ def contains(self, search_key: ScalarLike) -> ParentType:
             raise
         return res
 
-    def index(self, search_key: ScalarLike) -> ParentType:
-        search_key = cudf.Scalar(search_key)
+    def index(self, search_key: Union[ScalarLike, ColumnLike]) -> ParentType:
+        """
+        Return integers representing the index of the search key for each row.
+        If ``search_key`` is a sequence, it must be the same length as the
+        Series and ``search_key[i]`` represents the search key for the
+        ``i``-th row of the Series.
+
+        If the search key is not contained in a row, return -1.
+        If either the row or the search key are null, return <NA>.
```

Comment:
If the search key is contained multiple times, does this return the smallest matching index?
```suggestion
        If the search key is not contained in a row, return -1.
        If either the row or the search key are null, return <NA>.
        If the search key is contained multiple times, the smallest matching
        index is returned.
```